src/app/components/Sidebar.tsx

```diff
@@ -0,0 +1,157 @@
+"use client";
+
+import React from "react";
+import {
+  useSidebarStateContext,
+  useSidebarActionsContext,
+} from "@context/SidebarContext";
+import {
+  Drawer,
+  Box,
+  Typography,
+  List,
+  ListItemButton,
+  Divider,
+} from "@mui/material";
+import theme from "theme";
+import { getCourses } from "@services/client/course";
+import { useUserSession } from "@context/UserSessionContext";
+import { useRouter } from "next/navigation";
+import { usePathname } from "next/navigation";
+import { IdentifiableCourse } from "@interfaces/type";
+import MenuButton from "./buttons/MenuButton";
+import ModeEditOutlinedIcon from "@mui/icons-material/ModeEditOutlined";
+
+const Sidebar = React.memo(() => {
+  const [courses, setCourses] = React.useState<IdentifiableCourse[]>([]);
+  const { isOpen } = useSidebarStateContext();
+  const { closeSidebar } = useSidebarActionsContext();
+  const { user } = useUserSession();
+  const { courses: courseIds = [] } = user || {};
+  const router = useRouter();
+  const pathname = usePathname();
+
+  React.useEffect(() => {
+    const loadCourses = async () => {
+      const allCourses = await getCourses();
```

Comment:
We probably should do the filtering on database side.

Reply:
is that making a server component?

Reply:
The query to firebase should already filter to the course ids we're interested in.